app/scripts/app.js

```diff
@@ -378,8 +385,17 @@ angular
     $compileProvider.aHrefSanitizationWhitelist(/^\s*(https?|mailto|git):/i);
   })
   .run(function($rootScope, LabelFilter){
-    $rootScope.$on('$locationChangeSuccess', function(event) {
-      LabelFilter.setLabelSelector(new LabelSelector({}, true), true);
+    // assume we always want filterState persisted, pages that dont can turn it off
+    LabelFilter.persistFilterState(true);
+    LabelFilter.readPersistedState();
+    var initialLoad = true;
+    $rootScope.$on('$routeChangeSuccess', function(event) {
+      if (initialLoad) {
+        initialLoad = false;
+      }
+      else {
+        LabelFilter.setLabelSelector(new LabelSelector({}, true), true);
```

Comment:
Thats why it was switched to routeChangeSuccess instead of locationChangeSuccess.   routeChangeSuccess doesn't fire if you have reloadOnSearch:false